mlflow_yarn/yarn_backend.py

```diff
@@ -99,59 +108,74 @@ def run(self, project_uri: str, entry_point: str, params: Dict,
-                env=env
-            )
-            spec = skein.ApplicationSpec(services={"service": service})
-            app_id = self._skein_client.submit(spec)
-            MlflowClient().set_tag(active_run.info.run_id, "skein_application_id", app_id)
-            return YarnSubmittedRun(self._skein_client, app_id, active_run.info.run_id)
+        storage_dir = backend_config[PROJECT_STORAGE_DIR]
+
+        entry_point_command = project.get_entry_point(entry_point)\
+            .compute_command(params, storage_dir)
+
+        _logger.info(f"entry_point_command={entry_point_command}")
+
+        if project.conda_env_path:
+            spec_file = project.conda_env_path
+        else:
+            spec_file = os.path.join(work_dir, "requirements.txt")
+            if not os.path.exists(spec_file):
+                raise ValueError
+
+        package_path = cluster_pack.upload_spec(spec_file)
+        _logger.info(package_path)
+
+        additional_files = []
+        for file in os.listdir(work_dir):
+            full_path = os.path.join(work_dir, file)
+            if os.path.isfile(full_path):
+                additional_files.append(full_path)
+
+        entry_point, args = try_split_cmd(entry_point_command)
+
+        _logger.info(f"args {entry_point} {args}")
+
+        if "MLFLOW_YARN_TESTS" in os.environ:
+            # we need to have a real tracking server setup to be able to push the run id here
+            env = {"MLFLOW_TRACKING_URI": "file:/tmp/mlflow"}
+        else:
+            env = {
+                "MLFLOW_RUN_ID": active_run.info.run_id,
+                "MLFLOW_TRACKING_URI": mlflow.get_tracking_uri(),
+                "MLFLOW_EXPERIMENT_ID": experiment_id
+            }
+
+        _backend_dict = _get_backend_dict(work_dir)
+        # update config with what has been passed with --backend-config <json-new-config>
+        for key in _backend_dict.keys():
+            if key in backend_config:
+                _backend_dict[key] = backend_config[key]
+
+        _logger.info(f"backend config: {_backend_dict}")
+
+        app_id = skein_launcher.submit(
+                self._skein_client,
+                module_name=entry_point,
+                args=args,
+                package_path=package_path,
+                additional_files=additional_files,
+                env_vars=env,
+                process_logs=_upload_logs,
+                **_backend_dict)
+
+        MlflowClient().set_tag(active_run.info.run_id, "skein_application_id", app_id)
+        return YarnSubmittedRun(self._skein_client, app_id, active_run.info.run_id)
+
+
+def _get_backend_dict(work_dir: str) -> Dict:
+    backend_config = os.path.join(work_dir, "backend_config.json")
+    if os.path.exists(backend_config):
+        with open(backend_config, 'r') as f:
+            backend_config_dict = json.load(f)
+            if not isinstance(backend_config_dict, dict):
+                raise ValueError(f"{backend_config} file must be a dict")

```

Comment:
Here I think if the file is not a json, the json.load will fail and it will raise a JsonDecodeError, so it will never reach the if. Maybe you could use a try catch to catch this error instead.